client/tests/webdriver/specs/customFields.spec.js

```diff
@@ -0,0 +1,274 @@
+      CreateReport.waitForAlertToLoad()
+
+      expect(CreateReport.numberTrainedFieldShowed.getText()).to.not.include(
+        validNumberInput.toString()
+      )
+    })
+  })
+  // ------------------------------ PERSON CUSTOM FIELDS -----------------------------------------
+  describe("For person's custom fields", () => {
+    it("Should open create new person page", () => {
+      CreatePerson.openAsAdmin()
+      CreatePerson.form.waitForExist()
+      CreatePerson.form.waitForDisplayed()
+      // fill other required fields at the beginning
+      CreatePerson.lastName.setValue(REQUIRED_PERSON_FIELDS.lastname)
+      CreatePerson.rank.selectByAttribute("value", REQUIRED_PERSON_FIELDS.rank)
+      CreatePerson.gender.selectByAttribute(
+        "value",
+        REQUIRED_PERSON_FIELDS.gender
+      )
+    })
+
+    it("Should not show default invisible fields", () => {
+      CreatePerson.defaultInvisibleCustomFields.forEach(field => {
+        expect(field.isExisting()).to.eq(false)
+      })
+      // Date field is invisible by default
+      CreatePerson.addObjectButton.click()
+      CreatePerson.objectDateField.waitForExist({ reverse: true })
+    })
+
+    it("Should toggle correct invisible fields", () => {
+      // green toggles every invisible field to visible
+      CreatePerson.greenButton.click()
+      CreatePerson.defaultInvisibleCustomFields.forEach(field => {
+        field.waitForExist()
+      })
+      // Also it toggles array_of_objects date field
+      CreatePerson.objectDateField.waitForExist()
+    })
+
+    it("Should persist previous valid data when toggling field's visibility", () => {
+      CreatePerson.numberCustomField.setValue(validNumberInput)
+      // make default invisible fields invisible again, amber color does that
+      CreatePerson.amberButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist({ reverse: true })
+      // make visible
+      CreatePerson.greenButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist()
+
+      expect(CreatePerson.numberCustomField.getValue()).be.equal(
+        validNumberInput
+      )
+    })
+
+    it("Should not persist previous invalid data when toggling field's visibility", () => {
+      CreatePerson.numberCustomField.setValue(inValidNumberInput)
+      // Actually see the validation warning
+      CreatePerson.numberCustomFieldHelpText.waitForExist()
+      // make invisible
+      CreatePerson.amberButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist({ reverse: true })
+      // make visible
+      CreatePerson.greenButton.click()
+      CreatePerson.numberCustomFieldContainer.waitForExist()
+
+      expect(CreatePerson.numberCustomField.getValue()).not.be.equal(
+        inValidNumberInput
+      )
+      expect(CreatePerson.numberCustomField.getValue()).to.equal("")
+    })
+
+    it("Should validate visible field", () => {
+      CreatePerson.numberCustomField.setValue(inValidNumberInput)
```

Comment:
you are missing a test condition , you are only setting an invalid value here

Reply:
I am waitingForExist for helpText, which only exists when it is invalid. But i will also add text match as well.